packages/editor/src/components/rich-text/index.js

```diff
@@ -439,6 +440,7 @@ export class RichText extends Component {
 			}
 
 			this.setState( { start, end } );
+			this.props.onSelectionChange( this.props.clientId, this.props.identifier, { start, end } );
```

Comment:
This does require some investigation, because the local state in the RichText instances only track their own state. But if they use the global state the all track the state of the last RichText that had a selection change. I don't know what consequences this has.

To keep the scope from this PR increasing I've created https://github.com/WordPress/gutenberg/issues/11616 to track this.